onnx/defs/experiments/defs.cc

```diff
@@ -362,19 +361,21 @@ If scale is not provided, crop the borders as provided.)DOC")
 
 OPERATOR_SCHEMA(Embedding)
```

Comment:
Isn't it Gather

Reply:
Taking 2D weights as an example, 
Gather picks row by rowID, Embedding pick column by columnID. 

And yes, embedding can be done by transpose + gather + transpose, but it's very commonly used and deserves a separate op.

Reply:
Can you clarify when this op is actually useful in this setting? ONNX pretty much assumes row-major order so any sensible lookup table should be indexed by rows if it's to get any reasonable perf.

Reply:
Embedding is commonly used for NLP's word embedding lookup.

https://apple.github.io/coremltools/coremlspecification/sections/NeuralNetwork.html#embeddinglayerparams

Reply:
since it's in experimental mode, let's keep this. it's experimental mode anyways

Reply:
This isn't the same as Gather, embed multi the input with the embed matrix, it is only similar to gather if the input is in one hot presentation, right?

Reply:
Ok, seems like it's equivalent to Gather except that the weights matrix is transposed: it takes a list of indices and returns looks them up in the matrix. I'd argue that it never makes sense to keep matrix transposed (not sure why Apple did it this way) because of memory locality. Can't you just use Gather directly then?

Even though it's experimental I'd like to avoid duplication of operators with nearly-identical semantics

Reply:
Thank you on reviewing this. Could you please sign off?

Reply:
@dzhulgakov  if the input to embed isn't one hot vector, how this is gather?

Reply:
@dzhulgakov  As we have explained above. Embedding is not the same as Gather.